lib/seahorn/BvOpSem2.cc

```diff
@@ -895,12 +916,15 @@ class OpSemMemManager {
     for (unsigned i = 0; i < len; i += sem_word_sz) {
       Expr offset = bv::bvnum(i, ptrSzInBits(), m_efac);
       Expr dIdx = i == 0 ? dPtr : mk<BADD>(dPtr, offset);
-      unsigned word = 0;
+      unsigned long word = 0;
       for (int byte = sem_word_sz - 1; byte >= 0; byte--) {
-        word = word << 8;
+        word = word << 8L;
         if (i + byte < len)
           word += sPtr[i + byte];
       }
+      if (sem_word_sz == 8) {
+        word = (word << 32L) + (word >> 32L);
```

Comment:
comment on this code pattern. Why is addition `+` mixed with bit-vector operations?

Reply:
A bitwise or also works here. The idea is to swap the low 32 bits with the high 32 bits.